rslib/src/decks/tree.rs

```diff
@@ -256,8 +241,10 @@ impl Collection {
         let decks_map = self.storage.get_decks_map()?;
 
         add_collapsed_and_filtered(&mut tree, &decks_map, timestamp.is_none());
+
         if self.default_deck_is_empty()? {
-            hide_default_deck(&mut tree);
+            tree.children
+                .retain(|n| n.deck_id != 1 || !n.children.is_empty());
```

Comment:
The default deck may not be at the top level - you need to traverse the tree, like hide_default_deck() was doing.

Reply:
Add a check for the entire tree: [1dd6b36](https://github.com/ankitects/anki/pull/3492/commits/1dd6b3678303f8e33a6cddff376a1fac97dbeba8)